source/extensions/filters/http/health_check/health_check.h

```diff
@@ -17,11 +20,43 @@ namespace HttpFilters {
 namespace HealthCheck {
 
 /**
- * Shared cache manager used by all instances of a health check filter configuration as well as
- * all threads. This sets up a timer that will invalidate the cached response code and allow some
- * requests to go through to the backend. No attempt is made to allow only a single request to go
- * through to the backend, so during the invalidation window some number of requests will get
- * through.
+ * All health check filter stats. @see stats_macros.h
+ */
+#define ALL_HEALTH_CHECK_FILTER_STATS(COUNTER)                                                     \
+  COUNTER(request_total)                                                                           \
+  COUNTER(failed)                                                                                  \
+  COUNTER(ok)                                                                                      \
+  COUNTER(cached_response)                                                                         \
+  COUNTER(failed_cluster_not_found)                                                                \
+  COUNTER(failed_cluster_empty)                                                                    \
+  COUNTER(failed_cluster_unhealthy)                                                                \
+  COUNTER(degraded)
+
+/**
+ * Struct definition for all health check stats. @see stats_macros.h
+ */
+struct HealthCheckFilterStats {
+  ALL_HEALTH_CHECK_FILTER_STATS(GENERATE_COUNTER_STRUCT)
+
+  static HealthCheckFilterStats generateStats(const std::string& prefix, Stats::Scope& scope) {
+    const std::string final_prefix = absl::StrCat(prefix, "health_check.");
+    return {ALL_HEALTH_CHECK_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
+  }
+};
+
+// Forward declarations
+class HealthCheckFilter;
+class HealthCheckCacheManager;
+
+using HealthCheckCacheManagerSharedPtr = std::shared_ptr<HealthCheckCacheManager>;
+using HealthCheckFilterStatsSharedPtr = std::shared_ptr<HealthCheckFilterStats>;
+using ClusterMinHealthyPercentages = std::map<std::string, double>;
+using ClusterMinHealthyPercentagesConstSharedPtr =
+    std::shared_ptr<const ClusterMinHealthyPercentages>;
+using HeaderDataVectorSharedPtr = std::shared_ptr<std::vector<Http::HeaderUtility::HeaderDataPtr>>;
```

Comment:
If there's no strict need, I think the forward declarations should be removed, and using can be moved after the class declaration